pkg/virt-controller/watch/vm.go

```diff
@@ -929,25 +1039,79 @@ func (c *VMController) updateStatus(vm *virtv1.VirtualMachine, vmi *virtv1.Virtu
 	createdMatch := created == vm.Status.Created
 
 	ready := false
+
 	if created {
 		ready = controller.NewVirtualMachineInstanceConditionManager().HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady), k8score.ConditionTrue)
 	}
 	readyMatch := ready == vm.Status.Ready
 
-	if errMatch && createdMatch && readyMatch {
+	runStrategy, err := vm.RunStrategy()
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error getting RunStrategy: %v", err)
+	}
+	clearChangeRequest := false
+	if len(vm.Status.StateChangeRequests) != 0 {
+		// Only consider one stateChangeRequest at a time. The second and subsequent change
+		// requests have not been acted upon by this controller yet!
+		stateChange := vm.Status.StateChangeRequests[0]
+		if stateChange.Action == virtv1.StopRequest {
+			// If there's no VMI, then the VMI was stopped, and the stopRequest can be cleared
+			if vmi == nil {
+				log.Log.Object(vm).V(4).Infof("No VMI. Clearing stop request")
+				clearChangeRequest = true
```

Comment:
You will have to reconcile with the cluster directly when you want to pop something from the list. We have two sources which trigger updates to the controller loop. The VM and the VMI. If the VMI cache has issues, you may not see the VMI and pop the stop request because of an update to the VM in the informers. You will have to do an explicit GET request here if the cache says the the VMI is not there.